tests/test_api_callable.py

```diff
@@ -141,6 +141,27 @@ def test_call_kwargs(self):
         self.assertEqual(my_callable(None, CallOptions(key='updated')),
                          'updated')
 
+    def test_call_merge_options_metadata(self):
+        settings_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'val1'), ('key2', 'val2')]
+        }
+
+        settings = _CallSettings(kwargs=settings_kwargs)
+        my_callable = api_callable.create_api_call(
+            lambda _req, _timeout, **kwargs: kwargs, settings)
+
+        expected_kwargs = settings_kwargs
+        self.assertEqual(my_callable(None), expected_kwargs)
+
+        expected_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'updated_val1'), ('key2', 'val2')]
+        }
+        self.assertEqual(
+            my_callable(None, CallOptions(metadata=[('key1', 'updated_val1')])),
```

Comment:
The order _of the headers_ (e.g. `x-goog-api-client` vs. something else) does not matter; Python stores the headers internally as a dictionary.

The order _of the value of the header_ very much _does_ matter, and it is stored in GAX first as a `collections.OrderedDict` and then a string.